src/averageAges.js

```diff
@@ -38,6 +47,21 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const women = people.filter(person => person.sex === 'f');
+  const mothersNames = people.map(person => person.mother);
+  const mothersInList = women.filter(mothers =>
+    mothersNames.includes(mothers.name));
```

Comment:
you can simplify and filter the mothers with `withChildren` check inside the `filter` method